### PR TITLE
Sleep tests update for S3 and S0ix

### PR DESCRIPTION
**Pull request by Psotas**
d3fc1a68cac1f1521b02af9aea80c8a2a43fa47e will add workaround for https://github.com/Dasharo/open-source-firmware-validation/issues/129

### CODE REVIEW

File: lib/bios/menus.robot [hunk 1]
```diff
@@ -507,3 +507,40 @@ Boot System Or From Connected Disk
         ${system_index}=    Get Index Of Matching Option In Menu    ${menu_construction}    ${system_name}
     END
     Press Key N Times And Enter    ${system_index}    ${ARROW_DOWN}
+
+Check Platform Sleep Type Can Be Selected
```
Comment (macpijan) [inline lib/bios/menus.robot:511]: menus.robot is getting long already.

If we need specific keywords for specific menus, we may need another file, as we have for secure boot, for instance.

File: lib/bios/menus.robot [hunk 2]
```diff
@@ -507,3 +507,40 @@ Boot System Or From Connected Disk
...
+    ${platform_sleep_type_selectable}=    Run Keyword And Return Status
+    ...    Get Option State
+    ...    ${power_menu}
+    ...    Platform sleep type
+    Set Suite Variable    ${PLATFORM_SLEEP_TYPE_SELECTABLE}    ${platform_sleep_type_selectable}
+
+Set Platform Sleep Type
+    [Arguments]    ${platform_sleep_type}
+    Power On
+    IF    '${platform_sleep_type}' == 'S0ix'
```
Comment (macpijan) [inline lib/bios/menus.robot:525]: I would use different for these variables.

File: dasharo-compatibility/platform-hibernation-and-resume.robot [hunk 3]
```diff
@@ -37,10 +39,36 @@ HIB001.001 Cyclic platform hibernation and resume (Ubuntu 22.04)
     ...    and resume procedure performed cyclically works correctly
     Skip If    not ${HIBERNATION_AND_RESUME_SUPPORT}    HIB001.001 not supported
     Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    HIB001.001 not supported
+    Skip If    ${PLATFORM_SLEEP_TYPE_SELECTABLE}    HIB001.001 not supported
+    Cyclic Platform Hibernation And Resume (Ubuntu 22.04)
+
+HIB001.002 Cyclic platform hibernation and resume (Ubuntu 22.04) (S0ix)
```
Comment (macpijan) [inline dasharo-compatibility/platform-hibernation-and-resume.robot:45]: hibernation is not really related with the suspend mode AFAIK, @mkopec

File: dasharo-stability/m2-wifi.robot [hunk 4]
```diff
@@ -93,14 +94,40 @@ Suite Teardown      Run Keyword
...
 
-SMW0004.001 Wi-fi connection after suspension (Ubuntu 22.04)
+SMW004.001 Wi-fi connection after suspension (Ubuntu 22.04)
     [Documentation]    Check whether the Wi-Fi card is detected and working
     ...    correctly after performing suspension.
-    Skip If    not ${M2_WIFI_SUPPORT}    SUD004.001 not supported
-    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    SUD004.001 not supported
+    Skip If    not ${M2_WIFI_SUPPORT}    SMW004.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    SMW004.001 not supported
+    Skip If    ${PLATFORM_SLEEP_TYPE_SELECTABLE}    SMW004.001 not supported
```
Comment (macpijan) [inline dasharo-stability/m2-wifi.robot:102]: I do not get it, now we have 3 cases, where one is with S3, one with S0ix, one with what?

File: lib/bios/menus.robot (hunk 1 above)
Comment (Psotas) [reply lib/bios/menus.robot]: moved to `/lib/sleep-lib.robot` in e8394a4e17ed0293f131ab9bf39410ab7df5ae0f
Kwd name changed to `Check If Platform Sleep Type Can Be Selected`  https://github.com/Dasharo/open-source-firmware-validation/pull/128/files#diff-0d4b3b4e8afbeb5a9e8e3655f8d6ecf853f6e7e246a3da142cca93bd987e402eR6
Documentation added.

File: lib/bios/menus.robot (hunk 2 above)
Comment (Psotas) [reply lib/bios/menus.robot]: `${platform_sleep_type}`
Replaced by
`${platform_sleep_type_text}`
https://github.com/Dasharo/open-source-firmware-validation/pull/128/files#diff-0d4b3b4e8afbeb5a9e8e3655f8d6ecf853f6e7e246a3da142cca93bd987e402eR23

File: dasharo-compatibility/platform-hibernation-and-resume.robot (hunk 3 above)
Comment (Psotas) [reply dasharo-compatibility/platform-hibernation-and-resume.robot]: Ok, I deleted it.

File: lib/bios/menus.robot [hunk 5]
```diff
@@ -507,3 +507,40 @@ Boot System Or From Connected Disk
...
     END
     Press Key N Times And Enter    ${system_index}    ${ARROW_DOWN}
+
+Check Platform Sleep Type Can Be Selected
+    Power On
+    ${setup_menu}=    Enter Setup Menu Tianocore And Return Construction
+    ${dasharo_menu}=    Enter Dasharo System Features    ${setup_menu}
+    ${power_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Power Management Options
+    ${platform_sleep_type_selectable}=    Run Keyword And Return Status
+    ...    Get Option State
```
[the remark this answers is not in the document]
Comment (Psotas) [reply lib/bios/menus.robot]: In my opinion, this is the most readable.
Other options:
- use FOR loop for check menu construction from `${power_menu}` to check if there is a Platform sleep type
- Run Keyword and Return Status from `Get Index Of Matching Option In Menu`, but it less clear.

**Pull request merged**